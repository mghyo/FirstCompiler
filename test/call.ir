#start_function foo
int foo():
  return, 42, , 
#end_function foo

#start_function main
void main():
int-list: $t0
float-list: 
  callr,$t0,foo, ,
  call, printi, $t0
  return,,,
#end_function main
