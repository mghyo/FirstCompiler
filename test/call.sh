#!/bin/bash

set -e

./phase2 test/call.ir $1

diff out.s test/call.$1.s

spim -f out.s > tmp

diff tmp test/call.out

