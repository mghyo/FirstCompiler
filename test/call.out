Loaded: /usr/share/spim/exceptions.s
42